src/HotChocolate/Core/src/Types.Scalars/RegularExpressionType.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The Regular Expression scalar type represents a String Type following 
+    /// a regular expression pattern/format, as defined
+    /// here https://en.wikipedia.org/wiki/Regular_expression.
+    /// </summary>
+    public class RegularExpressionType : StringType
+    {
+        private readonly string _validationPattern;
+        private readonly Regex _validationRegex; 
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RegularExpressionType"/> class.
+        /// </summary>
+        public RegularExpressionType()
+            : this(
+                WellKnownScalarTypes.RegularExpression,
+                ScalarResources.RegularExpressionType_Description)
+        {
+        }
```

Comment:
This constructor is not needed, as without pattern the scalar is pretty much useless :)

Reply:
was not sure if the scalar types needed a parameter-less constructor, but I agree, if it is not the case...